wrappers/binaries.go

```diff
@@ -60,11 +135,28 @@ func AddSnapBinaries(s *snap.Info) (err error) {
 		}
 		created = append(created, wrapperPath)
 
-		if noCompletion || app.Completer == "" {
+		if completion == normalCompletion {
+			legacyCompPath := app.LegacyCompleterPath()
+			if dirs.IsCompleteShSymlink(legacyCompPath) {
+				os.Remove(legacyCompPath)
+			}
+		}
+
+		if completion == noCompletion || app.Completer == "" {
 			continue
 		}
 		// symlink the completion snippet
+		compDir := dirs.CompletersDir
+		if completion == legacyCompletion {
+			compDir = dirs.LegacyCompletersDir
+		}
+		if err := os.MkdirAll(compDir, 0755); err != nil {
+			return err
+		}
```

Comment:
nitpick: unless there is some harm in doing this, I'd move these lines outside of the `for` loop.

Reply:
We might not get to that point. For example if `app.Completer == ""` for all apps.